pydantic/_internal/_mock_val_ser.py

```diff
@@ -109,43 +110,74 @@ def rebuild(self) -> ValSer | None:
         return None
 
 
-def set_type_adapter_mocks(adapter: TypeAdapter, type_repr: str) -> None:
-    """Set `core_schema`, `validator` and `serializer` to mock core types on a type adapter instance.
+def _attempt_rebuild_fn(
+    cls_: T, attr_name: str, rebuild_fn: Callable[[T], Callable[..., bool | None]]
+) -> Callable[[], Any | None]:
+    """Attempt to rebuild the core schema associated with cls_ and return the new value of the core schema attr, if successful.
 
-    Args:
-        adapter: The type adapter instance to set the mocks on
-        type_repr: Name of the type used in the adapter, used in error messages
+    Ex: this function could call cls_.model_rebuild() if `cls_` were a `BaseModel` subclass, then attempt to fetch the value of `cls_.__pydantic_core_schema__`.
     """
-    undefined_type_error_message = (
-        f'`TypeAdapter[{type_repr}]` is not fully defined; you should define `{type_repr}` and all referenced types,'
-        f' then call `.rebuild()` on the instance.'
-    )
 
-    def attempt_rebuild_fn(attr_fn: Callable[[TypeAdapter], T]) -> Callable[[], T | None]:
-        def handler() -> T | None:
-            if adapter.rebuild(_parent_namespace_depth=5) is not False:
-                return attr_fn(adapter)
-            else:
-                return None
+    def handler() -> Any | None:
+        if rebuild_fn(cls_)(raise_errors=False, _parent_namespace_depth=5) is not False:
+            return getattr(cls_, attr_name)
+        return None
+
+    return handler
 
-        return handler
 
-    adapter.core_schema = MockCoreSchema(  # pyright: ignore[reportAttributeAccessIssue]
-        undefined_type_error_message,
+class CoreAttrLookup(TypedDict):
+    core_schema: str
+    validator: str
+    serializer: str
+
+
+def _set_mocks(
+    cls_: T, rebuild_fn: Callable[[T], Callable[..., bool | None]], core_attr_lookup: CoreAttrLookup, error_message: str
```

Comment:
Maybe it could refractor the error formatting at the same time. Eg doing the error message forming on demand. Allowing more "heavier" formatting